tensorflow/core/common_runtime/sycl/sycl_allocator.h

```diff
@@ -46,8 +50,14 @@ class SYCLAllocator : public Allocator {
   // AllocatedSize(void* ptr) by default.
   size_t RequestedSize(void* ptr) override;
   Eigen::SyclDevice* getSyclDevice() { return sycl_device_; }
+  // Clear the SYCL device used by the Allocator
+  void ClearSYCLDevice() {
+    delete sycl_device_;
```

Comment:
Should this be protected by a mutex. Might it now race with some of the if (sycl_device_) usage above?